patches/chromium/feat_add_support_for_overriding_the_base_spellchecker_download_url.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <sattard@slack-corp.com>
+Date: Tue, 25 Feb 2020 13:28:30 -0800
+Subject: feat: add support for overriding the base spellchecker download URL
+
+This patch is required as the testing-only method we were using does not
+take into account the dictionary name and therefore will not work for
+production use cases.  This is unlikely to be upstreamed as the change
+is entirely in //chrome.
+
+diff --git a/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc b/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
+index 0fb84989f9b2f84ec2a2589dda3d91cf59c0adda..71d6ce58f7340803c78b8310d4e95aa18041ad56 100644
+--- a/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
++++ b/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
+@@ -48,6 +48,9 @@ namespace {
+ base::LazyInstance<GURL>::Leaky g_download_url_for_testing =
+     LAZY_INSTANCE_INITIALIZER;
+ 
++base::LazyInstance<GURL>::Leaky g_base_download_url_override =
++    LAZY_INSTANCE_INITIALIZER;
++
+ // Close the file.
+ void CloseDictionary(base::File file) {
+   base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
+@@ -247,6 +250,10 @@ void SpellcheckHunspellDictionary::SetDownloadURLForTesting(const GURL url) {
+   g_download_url_for_testing.Get() = url;
+ }
+ 
++void SpellcheckHunspellDictionary::SetBaseDownloadURL(const GURL url) {
++  g_base_download_url_override.Get() = url;
++}
++
+ GURL SpellcheckHunspellDictionary::GetDictionaryURL() {
+   if (g_download_url_for_testing.Get() != GURL())
+     return g_download_url_for_testing.Get();
+@@ -254,8 +261,11 @@ GURL SpellcheckHunspellDictionary::GetDictionaryURL() {
+   std::string bdict_file = dictionary_file_.path.BaseName().MaybeAsASCII();
+   DCHECK(!bdict_file.empty());
+ 
++  if (g_base_download_url_override.Get() != GURL())
++    return GURL(g_base_download_url_override.spec() + base::ToLowerASCII(bdict_file));
++
+   static const char kDownloadServerUrl[] =
+-      "https://redirector.gvt1.com/edgedl/chrome/dict/";
++     "https://redirector.gvt1.com/edgedl/chrome/dict/";
```

Comment:
this change seems spurious...?

Reply:
yeah noticed that after push, got rid of it 👍